config/src/main/java/com/typesafe/config/impl/ConfigNodeKeyValue.java

```diff
@@ -0,0 +1,55 @@
+/**
+ *   Copyright (C) 2015 Typesafe Inc. <http://typesafe.com>
+ */
+package com.typesafe.config.impl;
+
+import com.typesafe.config.ConfigException;
+import com.typesafe.config.ConfigNode;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+final class ConfigNodeKeyValue extends AbstractConfigNode {
+    final private ArrayList<AbstractConfigNode> children;
+
+    public ConfigNodeKeyValue(Collection<AbstractConfigNode> children) {
+        this.children = new ArrayList(children);
+    }
+
+    protected Collection<Token> tokens() {
+        ArrayList<Token> tokens = new ArrayList();
+        for (AbstractConfigNode child : children) {
+            tokens.addAll(child.tokens());
+        }
+        return tokens;
+    }
+
+    public ConfigNodeKeyValue replaceValue(AbstractConfigNodeValue newValue) {
+        ArrayList<AbstractConfigNode> childrenCopy = new ArrayList(children);
+        for (int i = 0; i < childrenCopy.size(); i++) {
+            if (childrenCopy.get(i) instanceof AbstractConfigNodeValue) {
+                childrenCopy.set(i, newValue);
+                return new ConfigNodeKeyValue(childrenCopy);
+            }
+        }
+        throw new ConfigException.BugOrBroken("KeyValue node doesn't have a value");
+    }
+
+    public AbstractConfigNodeValue value() {
+        for (int i = 0; i < children.size(); i++) {
+            if (children.get(i) instanceof AbstractConfigNodeValue) {
+                return (AbstractConfigNodeValue)children.get(i);
+            }
+        }
+        throw new ConfigException.BugOrBroken("KeyValue node doesn't have a value");
+    }
+
+    public ConfigNodeKey key() {
```

Comment:
this should probably be called path() and the KeyValue class could either be called `PathValue` or perhaps simply `Field` would be easier.